res_partner_operating_unit/views/res_partner_view.xml

```diff
@@ -4,15 +4,14 @@
     License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl-3.0) -->
     <!-- Partner Form View  -->
     <record model="ir.ui.view" id="view_partner_form">
-        <field name="name">res.partner.form</field>
         <field name="model">res.partner</field>
         <field name="inherit_id" ref="base.view_partner_form" />
         <field name="arch" type="xml">
             <field name="category_id" position="after">
                 <field
                     name="operating_unit_ids"
                     groups="operating_unit.group_multi_operating_unit"
-                    domain="[('company_id','=', company_id)]"
+                    domain="[('company_id', '=', company_id)]"
```

Comment:
I think this domain can be replaced with a `_check_company_auto` at the partner model level

Odoo is doing the same but in the stock module instead of the base module. https://github.com/odoo/odoo/blob/5dafc19f72116f2784c0308b5df40f0ec3f117a9/addons/stock/models/res_partner.py#L10